### PR TITLE
wormchain: run Ignite through docker and update documentation

### PR DESCRIPTION
**Pull request by nik-suri**
To avoid system compatibility issues, it'd be better to run ignite commands through docker.

This PR also updates documentation that previously referred to starport and updates the wormchain CI pipeline to check diffs in generated protobuf files.

### CODE REVIEW

File: wormchain/Makefile [hunk 1]
```diff
@@ -1,6 +1,7 @@
 PROTO_FILES=$(shell find proto -name "*.proto")
 GO_FILES=$(shell find . -name "*.go")
-IGNITE_EXPECTED_VERSION:=v0.23.0
+PARENT_DIR=$(shell cd ../ && pwd)
+IGNITE_CONTAINER=ignitehq/cli:0.23.0
```
Comment (hendrikhofstadt) [inline wormchain/Makefile:4]: pin the container hash

**Review by hendrikhofstadt — CHANGES_REQUESTED** [review]

File: wormchain/README.md [hunk 2]
```diff
@@ -16,18 +16,18 @@ See [development.md](./development.md)
 
 ## How to run the tests
 
-    run either "starport chain serve" or "tilt up"
+    run "tilt up"
```
Comment (jynnantonix) [inline wormchain/README.md:19]: I think tilt won't run wormchain by default so this probably should be `tilt up -- --wormchain`.

**Review by jynnantonix — DISMISSED** [review]

**Review by conorpp — APPROVED** [review]

File: wormchain/Dockerfile.proto [hunk 3]
```diff
@@ -0,0 +1,18 @@
+FROM docker.io/golang:1.19.3@sha256:dc76ef03e54c34a00dcdca81e55c242d24b34d231637776c4bb5c1a8e8514253 AS go-build
```
Comment (jynnantonix) [inline wormchain/Dockerfile.proto:1]: Why not use the `ignitehq/cli:0.23.0` docker image?  It comes with go installed and then we don't need to fetch the ignite cli separately.

Comment (nik-suri) [reply wormchain/Dockerfile.proto:1]: I wasn't able to run the following commands in the ignite docker image

File: wormchain/development.md [hunk 4]
```diff
@@ -44,7 +43,7 @@ Golang tests
 
     make test
 
-Client tests, run against the chain. Wormchain must be running via `starport chain serve`, `make run` or `tilt up`
+Client tests, run against the chain. Wormchain must be running via `make run` or `tilt up`
```
Comment (hendrikhofstadt) [inline wormchain/development.md:46]: tilt up --wormchain

File: wormchain/Dockerfile.proto [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+RUN mkdir /app
+
+COPY wormchain /app/wormchain
+COPY sdk /app/sdk
+
+# download ignite and build proto
+RUN wget https://github.com/ignite/cli/releases/download/v0.23.0/ignite_0.23.0_linux_amd64.tar.gz
```
Comment (hendrikhofstadt) [inline wormchain/Dockerfile.proto:9]: we need to pin the hash here

**Review by hendrikhofstadt — CHANGES_REQUESTED** [review]
See inline comments

File: wormchain/Dockerfile.proto (hunk 5 above)
Comment (nik-suri) [reply wormchain/Dockerfile.proto:9]: added line to validate checksum below

File: wormchain/Dockerfile.proto (hunk 3 above)
Comment (nik-suri) [reply wormchain/Dockerfile.proto]: Thanks @jynnantonix. Using this Dockerfile

File: wormchain/Makefile [hunk 6]
```diff
@@ -44,17 +43,13 @@ validators:
...
 	go build -v $(BUILD_FLAGS) -tags ledger -o $@ $<
 
-check-ignite:
-	@ if [ "$(shell ignite version | awk '/Ignite CLI version:/ { print $$4 }')" != "$(IGNITE_EXPECTED_VERSION)" ] ; then echo "Expected ignite version $(IGNITE_EXPECTED_VERSION)" && exit 1 ; fi
-
-proto: check-ignite $(PROTO_FILES)
-	ignite generate proto-go
-	touch proto
+proto: $(PROTO_FILES)
+	cd .. && docker build --target go-export -f wormchain/Dockerfile.proto -o type=local,dest=wormchain .
```
Comment (evan-gray) [inline wormchain/Makefile:47]: do you need to `cd ..` as opposed to 
```
docker build --target go-export -f Dockerfile.proto -o type=local,dest=. ..
```

**Review by hendrikhofstadt — APPROVED** [review]

File: wormchain/Makefile (hunk 6 above)
Comment (nik-suri) [reply wormchain/Makefile:47]: we need to `cd ..`. `docker build` only has access to the directories in the context it is run.

Comment (evan-gray) [reply wormchain/Makefile:47]: Yes, I am suggesting you run it from the context of `..`

Comment (nik-suri) [reply wormchain/Makefile:47]: My bad the above does work.. will update PR.

**Review by jynnantonix — DISMISSED** [review]

**Review by hendrikhofstadt — APPROVED** [review]

File: wormchain/Makefile [hunk 7]
```diff
@@ -44,17 +43,13 @@ validators:
...
 	go build -v $(BUILD_FLAGS) -tags ledger -o $@ $<
 
-check-ignite:
-	@ if [ "$(shell ignite version | awk '/Ignite CLI version:/ { print $$4 }')" != "$(IGNITE_EXPECTED_VERSION)" ] ; then echo "Expected ignite version $(IGNITE_EXPECTED_VERSION)" && exit 1 ; fi
-
-proto: check-ignite $(PROTO_FILES)
-	ignite generate proto-go
-	touch proto
+proto: $(PROTO_FILES)
+	docker build --target go-export -f Dockerfile.proto -o type=local,dest=wormchain ..
```
Comment (evan-gray) [inline wormchain/Makefile:47]: `dest=.`, this will result in generated files going into `wormchain/wormchain/x/wormhole/types`

**Review by evan-gray — CHANGES_REQUESTED** [review]

**Review by evan-gray — APPROVED** [review]

**Review by jynnantonix — APPROVED** [review]

**Review by hendrikhofstadt — APPROVED** [review]

**Pull request merged**